hack/generator/pkg/astmodel/armconversion/convert_from_arm_function_builder.go

```diff
@@ -0,0 +1,565 @@
+					X:   builder.receiverIdent,
+					Sel: ast.NewIdent(string(toProp.PropertyName())),
+				},
+				token.ASSIGN,
+				ast.NewIdent("owner"))
+			return []ast.Stmt{result}
+		},
+	}
+}
+
+func (builder *convertFromArmBuilder) propertiesWithSameNameAndTypeHandler() propertyConversionHandler {
+
+	return propertyConversionHandler{
+		finder: func(toProp *astmodel.PropertyDefinition, fromType *astmodel.ObjectType) (bool, *astmodel.PropertyDefinition) {
+
+			fromProp, ok := fromType.Property(toProp.PropertyName())
+			if ok && toProp.PropertyType().Equals(fromProp.PropertyType()) {
+				return true, fromProp
+			}
+			return false, nil
+		},
+		matchHandler: func(toProp *astmodel.PropertyDefinition, fromProp *astmodel.PropertyDefinition) []ast.Stmt {
+
+			result := astbuilder.SimpleAssignment(
+				&ast.SelectorExpr{
+					X:   builder.receiverIdent,
+					Sel: ast.NewIdent(string(fromProp.PropertyName())),
+				},
+				token.ASSIGN,
+				&ast.SelectorExpr{
+					X:   builder.typedInputIdent,
+					Sel: ast.NewIdent(string(toProp.PropertyName())),
+				})
+			return []ast.Stmt{result}
+		},
+	}
+}
+
+func (builder *convertFromArmBuilder) propertiesWithSameNameButDifferentTypeHandler() propertyConversionHandler {
+	temp := false
+	definedErrVar := &temp
+
+	return propertyConversionHandler{
+		finder: func(toProp *astmodel.PropertyDefinition, fromType *astmodel.ObjectType) (bool, *astmodel.PropertyDefinition) {
+			fromProp, ok := fromType.Property(toProp.PropertyName())
+			if ok && !toProp.PropertyType().Equals(fromProp.PropertyType()) {
+				return true, fromProp
+			}
+			return false, nil
+		},
+		matchHandler: func(toProp *astmodel.PropertyDefinition, fromProp *astmodel.PropertyDefinition) []ast.Stmt {
+			var result []ast.Stmt
+
+			if !*definedErrVar {
+				result = append(
+					result,
+					astbuilder.SimpleVariableDeclaration(ast.NewIdent("err"), ast.NewIdent("error")))
+				*definedErrVar = true
+			}
+
+			complexConversion := builder.fromArmComplexPropertyConversion(
+				&ast.SelectorExpr{
+					X:   builder.typedInputIdent,
+					Sel: ast.NewIdent(string(fromProp.PropertyName())),
+				},
+				&ast.SelectorExpr{
+					X:   builder.receiverIdent,
+					Sel: ast.NewIdent(string(toProp.PropertyName())),
+				},
+				toProp.PropertyType(),
+				string(toProp.PropertyName()),
+				nil,
+				nil)
+
+			result = append(result, complexConversion...)
+			return result
+		},
+	}
+}
+
+//////////////////////////////////////////////////////////////////////////////////
+// Complex property conversion (for when properties aren't simple primitive types)
+//////////////////////////////////////////////////////////////////////////////////
+
+func (builder *convertFromArmBuilder) fromArmComplexPropertyConversion(
+	source ast.Expr,
+	destination ast.Expr,
+	destinationType astmodel.Type,
+	nameHint string,
+	conversionContext []astmodel.Type,
+	assignmentHandler func(result ast.Expr, destination ast.Expr) ast.Stmt) []ast.Stmt {
```

Comment:
Will do this -- since the signatures are actually the same for to/from (I did that on purpose), using a shared type goes a bit towards showing that the expectation is these two things look similar (even though we can't use the same functions due to the asymmetry I mentioned earlier)

Reply:
Ah, although it does fuss with the specific type handling functions a bit (they need an extra cast in them now, which is unfortunate).

Still I think probably worth doing...

Reply:
I've done this in its own commit so if we don't like the look we can back it out. Check it out and see what you think.